src/main/java/com/sahabuddin/contactmanager/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package com.sahabuddin.contactmanager.security;
+
+import com.sahabuddin.contactmanager.entities.User;
+import com.sahabuddin.contactmanager.respositories.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+        User user = userRepository.findByEmail(username);
```

Comment:
Email can be case-insensitive. So, it is better to use findByEmail with ignore case.